### ISSUE TITLE
Dark Mode on macos

### ISSUE THREAD

**Issue by jonathanhirz** [defold/defold#4808]
![doesn't that dark mode look amazing?](https://user-images.githubusercontent.com/1631578/82518633-a62e0700-9ad4-11ea-8134-c12002e4bab5.png)

> "wow, that dark mode menu bar looks great!"

Hello! I haven't used defold in a while. The new editor looks great. I would love to be able to keep the top menu bar dark in dark mode on macos. I have done this temporarily in other apps, according to this aseprite post:
https://community.aseprite.org/t/mac-aseprite-1-2-9-theme-window-color/2027/5

You need this key:
`<key>NSRequiresAquaSystemAppearance</key>
<false/>`

in the app's info.plist file. If I change it on my system, sometimes the app doesn't launch. I don't know why it works sometimes and not others, but I think it needs to be build with newer Xcode or something. Would be great if this could be added in at some time, but not a priority. Thanks!

**Comment by defold-services** [comment]
Thanks @jonathanhirz. I have prepared a PR so that we can get CI to build an editor that we can test.

It tried and it seemed to for on my version of macOS at least (Catalina 10.15.4). Could you please give it a try as well:

https://d.defold.com/archive/d32d58d868a7403ce23034f8b4f5c0e62bbcee3e/stable/editor2/Defold-x86_64-darwin.dmg

**Comment by jonathanhirz** [comment]
Looks great! I'm on the same version of macOS. Thank you so much, that was very fast work =]

**Comment by jonathanhirz** [comment]
Sorry to bump this again, but I'm curious about the build procedure. It seems like this fix is already in the downloads from the website. Does "CI" regularly build versions with these pull requests/fixes and update the version that's available? When do version numbers change? Am I just seeing some weird behavior now and I'm just getting the dark mode all the time? I do see the key in the plist file, so I assume I'm right. Just curious. Also, do I close this or does someone else manage that?
Thanks again!

**Comment by JCash** [comment]
Fixes like this, that we do in the editor, we commit to the `editor-dev` branch. The CI builds and releases that branch automatically.
Once that's released, your old editor checks if there's an update to the same channel (`editor-alpha` in this case). If there _is_ an update, it will display a blue link at the bottom right of the editor saying "Update" (or something). If you click that, it downloads the new editor, then asks you if you wish to restart.

The editor sha1 should change, the engine sha1 should not (we release a new engine every 4 weeks)

**Comment by britzl** [comment]
> Also, do I close this or does someone else manage that?

Someone (you or us) should close it once we've verified that it is working :-)

It was added to the current 1.2.170 project/sprint so that we can keep track of it, and at the end of the dev cycle we will close all issues that have been fixed. But there is no harm done if you had closed it before that either.

**Comment by AGulev** [comment]
I am not sure if this fix merged or not, I saw it was reverted, right?

**Comment by jonathanhirz** [comment]
> I am not sure if this fix merged or not, I saw it was reverted?

Can confirm, back to light mode menu bar today.

**Comment by britzl** [comment]
Yes, we had to revert this as there was problems on older macOS versions.

**Comment by AGulev** [comment]
Related: https://github.com/defold/editor2-issues/issues/2209

**Comment by thejustinwalsh** [comment]
@britzl is there any context to the problem with an older macOS version that was experienced?

If it was blurry rendering on a HiDPI (builtin display)
https://git.libreoffice.org/core/+/645fe53be0dc36535dba0ed684e21ca4cda80d70%5E%21

Setting `LC_VERSION_MIN_MACOSX` to `0.0` when building the app seems to be the hack that is needed to magically fix it.

Found this from the SDL post about dark mode: https://github.com/libsdl-org/SDL/issues/3637

**Comment by thejustinwalsh** [comment]
Here is the local workaround for those who want darkmodetitle bar in the meantime

Apply Dark Mode:
```sh
defaults write com.defold.editor NSRequiresAquaSystemAppearance -bool NO
# might need this too if above doesn't work
defaults write net.java.openjdk.cmd NSRequiresAquaSystemAppearance -bool NO
```

Revert:
```sh
defaults delete com.defold.editor NSRequiresAquaSystemAppearance
# might need this too if above doesn't work
defaults delete net.java.openjdk.cmd NSRequiresAquaSystemAppearance
```

**Comment by britzl** [comment]
> If it was blurry rendering on a HiDPI (builtin display)

I believe it was right-click that stopped working(!)

**Comment by thejustinwalsh** [comment]
Ahh, weird. Ok. Setting the defaults is a nice workaround until either, the old version of macOS is fully depreciated or a new Xcode solves this apple bug.

**Comment by gianmichelemariani** [comment]
Any chance this can go into the newest build? It has been quite a while and we're now at Monterey on macs

**Comment by britzl** [comment]
@gianmichelemariani: There's a build for you to test here: http://d.defold.com/archive/dev/5cccb19908dadf6ce4fbeda5ca6e10b1421e9b06/dev/editor2/Defold-x86_64-darwin.dmg

**Comment by britzl** [comment]
> I believe it was right-click that stopped working(!)

More info here: https://forum.defold.com/t/right-click-not-working-in-defold-editor/65308/23

**Comment by gianmichelemariani** [comment]
@britzl I've been using it for a couple of days and the build seems to work fine although the colour is still on the bright side compared to other applications. Do you have control over it?

<img width="1060" alt="Screenshot 2021-12-16 at 18 57 25" src="https://user-images.githubusercontent.com/46694168/146432442-2d292254-9227-42c3-a5be-9d5144e3882b.png">

<img width="476" alt="Screenshot 2021-12-16 at 19 00 26" src="https://user-images.githubusercontent.com/46694168/146432465-56bd2258-dae1-4da7-914d-821900b6bbe2.png">

**Comment by britzl** [comment]
> Do you have control over it?

No, I don't think so. But I believe the menu bar is slightly translucent and it might be that there's an option to set a colour that might be shining through the menu bar and making it look a bit brighter.
